Jim-Hodapp-Coaching/refactor-platform-fe
# Add more advanced text editor for coaching notes

## Description
This PR adds a new CoachingNotes and TipTapEditor pair of components to bring rich text editing into a coaching session.


#### GitHub Issue: #42 

### Changes
* Adds a new `TipTapEditor` component which wraps the TipTap rich text editor
* Enables a nice baseline of desirable rich text editing functionality
* Stores coaching notes as a TipTap JSON document type
* Moves `CoachingNotes` into a separate file and out of coaching session page.tsx
* Adds Sass styling sheets for TipTap and its child components
* Can use Tailwindcss tags and classes in Sass stylesheets

### Screenshots / Videos Showing UI Changes
![Jim __ Caleb _ Oct 14_ 2024](https://github.com/user-attachments/assets/0b5292b0-6ec7-4774-984e-8e99e503d26c)

### Testing Strategy
1. Navigate to the coaching session page
2. Start editing text in the coaching notes text area
3. Exercise all text formatting options in the toolbar
4. Also try toggling light/dark mode and ensure all elements remain visible
5. Navigate away from this session and back into it
6. Notice that the rich text formatting is restored on page load

### Concerns
None
